### PR TITLE
Olga/927 normalize operator values

### PR DESCRIPTION
**Pull request by okavvada**
Connects to #927.

- [x] Update tests
- [ ] Update data migrations
- [x] Update docs

No need to update migrations.

@nvkelso , @rmarianski , @zerebubuth could you please review?

### CODE REVIEW

File: integration-test/927-normalize-operator-values.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+# Department of National Parks NSW in landuse
+# http://www.openstreetmap.org/way/429280600
+assert_has_feature(
+    16, 60128, 39504, 'landuse',
+    {'id': 429280600, 'operator': 'National Parks & Wildife Service NSW'})
+
+assert_no_matching_feature(
+    16, 60128, 39504, 'landuse',
+    {'id': 429280600, 'operator': 'Department of National Parks NSW'})
```
Comment (nvkelso) [inline integration-test/927-normalize-operator-values.py]: Where does this feature occur, is it in the USA or Australia? (New South Wales?)

Comment (okavvada) [inline integration-test/927-normalize-operator-values.py]: This one is in Australia. Do we want to keep it US-centric? While I was looking for tags for `National Parks` and `Forest Service`, I came across the NSW ones and made a separate transform for those. They do not get merged with the US ones.

Comment (nvkelso) [inline integration-test/927-normalize-operator-values.py]: Yep, NSW so be careful.

File: vectordatasource/transform.py [hunk 2]
```diff
@@ -3622,3 +3622,64 @@ def simplify_and_clip(ctx):
...
+    'U.S. National Park Service', 'US National Park Service' etc that refer
+    to the same operator tag. This function promotes a normalized value
+    for all alternatives in specific operator values.
+
+    See https://github.com/tilezen/vector-datasource/issues/927.
+    """
+
+    operator = properties.get('operator', None)
+    if operator is not None:
+        if operator in ('National Park Service',
```
Comment (nvkelso) [inline vectordatasource/transform.py]: Be careful of general services / parks. Might try adding a bbox check?

File: vectordatasource/transform.py [hunk 3]
```diff
@@ -3622,3 +3622,64 @@ def simplify_and_clip(ctx):
...
+                        'United States National Park Service',
+                        'US National Park Service',
+                        'U.S. National Park Service',
+                        'National Park Services',
```
Comment (nvkelso) [inline vectordatasource/transform.py]: Be careful of general services / parks. Might try adding a bbox check?

File: vectordatasource/transform.py [hunk 4]
```diff
@@ -3622,3 +3622,64 @@ def simplify_and_clip(ctx):
...
+                        'US National Park service',
+                        'National Parks and Wildlife Service',
```
Comment (nvkelso) [inline vectordatasource/transform.py]: Be careful of general services / parks. Might try adding a bbox check?

Also, I've never heard of this one – is it USA only?

File: vectordatasource/transform.py [hunk 5]
```diff
@@ -3622,3 +3622,64 @@ def simplify_and_clip(ctx):
...
+                        'National Parks',
```
Comment (nvkelso) [inline vectordatasource/transform.py]: Be careful of general services / parks. Might try adding a bbox check?

File: vectordatasource/transform.py [hunk 6]
```diff
@@ -3622,3 +3622,64 @@ def simplify_and_clip(ctx):
...
+                        'National Park'):
```
Comment (nvkelso) [inline vectordatasource/transform.py]: Be careful of general services / parks. Might try adding a bbox check?

Also, I've never heard of this one – is it USA only?

File: vectordatasource/transform.py [hunk 7]
```diff
@@ -3622,3 +3622,64 @@ def simplify_and_clip(ctx):
...
+            properties['operator'] = 'National Park Service'
+
+            return (shape, properties, fid)
+
+        if operator in ('United States Forest Service',
+                        'US Forest Service',
+                        'U.S. Forest Service',
+                        'USDA Forest Service',
+                        'National Forest Service',
```
Comment (nvkelso) [inline vectordatasource/transform.py]: Careful of generics.

File: vectordatasource/transform.py [hunk 8]
```diff
@@ -3622,3 +3622,64 @@ def simplify_and_clip(ctx):
...
+                        'United States Department of Agriculture',
+                        'Forest Service',
```
Comment (nvkelso) [inline vectordatasource/transform.py]: Careful of generics.

File: vectordatasource/transform.py [hunk 9]
```diff
@@ -3622,3 +3622,64 @@ def simplify_and_clip(ctx):
...
+                        'NSW National Parks and Wildlife Service',
+                        'NSW Parks and Wildlife Service',
+                        'NSW Parks and Wildlife Service (NPWS)',
+                        'National Parks & Wildife Service NSW',
+                        'National Parks and WIldlife Service NSW',
+                        'National Parks and Wildlife NSW',
+                        'National Parks and Wildlife Service NSW',
+                        'National Parks and Wildlife',
+                        'National Parks and Wildlife Service'):
+            properties['operator'] = 'National Parks & Wildife Service NSW'
```
Comment (nvkelso) [inline vectordatasource/transform.py]: Some of these smell like different things? How does `National Parks & Wildife Service NSW` describe itself outside of OSM?

File: integration-test/927-normalize-operator-values.py (hunk 1 above)
Comment (nvkelso) [inline integration-test/927-normalize-operator-values.py]: Ah, you have a 3rd transform for Australia, this is okay then.

File: vectordatasource/transform.py (hunk 9 above)
Comment (okavvada) [inline vectordatasource/transform.py]: `NSW National Parks` and `NSW Wildlife service` go to the same website so I figured they are the same entity.

Comment (nvkelso) [inline vectordatasource/transform.py]: Namely: https://www.nationalparks.nsw.gov.au where their logo includes both terms. 

LGTM

File: docs/layers.md [hunk 10]
```diff
@@ -448,6 +448,8 @@ Points of interest from OpenStreetMap, with per-zoom selections similar to the p
 
 Features from OpenStreetMap which are tagged `disused=*` for any other value than `disused=no` are not included in the data. Features which have certain parenthetical comments after their name are suppressed until zoom 17 and have their `kind` property set to that comment. Currently anything with a name ending in '(closed)' or '(historical)' will be suppressed in this manner. Railway stops, halts, stations and tram stops from OpenStreetMap tagged with a `historic` tag are also not included in the data.
 
+To promote consistency in data tagging we merged all relevant operator tags for `US National Parks`, under the common tag `United States National Park Service`, for `US Forest Service` under the common tag `United States Forest Service` and all North South Wales National Parks under the common tag `National Parks & Wildife Service NSW`. Generic tags such as `National Park`, `National Forest Service` and `National Parks & Wildife Service` are exported seperately.
```
Comment (nvkelso) [inline docs/layers.md]: Revise to fix a few typos and removing the generic tags section:

To resolve inconsistency in data tagging in OpenStreetMap we normalize several operator values for United States National Parks as  `United States National Park Service`, several United States Forest Service values as `United States Forest Service`, and several values for New South Wales National Parks in Australia as `National Parks & Wildlife Service NSW`.

File: vectordatasource/transform.py (hunk 2 above)
Comment (nvkelso) [inline vectordatasource/transform.py]: Resolved in person, all of these are in the USA (and there are hundreds of them), so okay to leave as-is.

**Comment by nvkelso** [comment]
Fix documentation in https://github.com/tilezen/vector-datasource/pull/929#discussion_r73022582 and then I think it's complete?

File: vectordatasource/transform.py [hunk 11]
```diff
@@ -3622,3 +3622,56 @@ def simplify_and_clip(ctx):
...
+                        'NSW Parks and Wildlife Service (NPWS)',
+                        'National Parks & Wildife Service NSW',
+                        'National Parks and WIldlife Service NSW',
+                        'National Parks and Wildlife NSW',
+                        'National Parks and Wildlife Service NSW'):
+            properties['operator'] = 'National Parks & Wildife Service NSW'
+
+            return (shape, properties, fid)
+
+    return (shape, properties, fid)
```
Comment (rmarianski) [inline vectordatasource/transform.py]: Would this be better implemented as a lookup table instead?

Comment (nvkelso) [inline vectordatasource/transform.py]: Yes, that's more in keeping with some of the other transforms.

File: vectordatasource/transform.py [hunk 12]
```diff
@@ -3622,3 +3622,58 @@ def simplify_and_clip(ctx):
...
+                flattened.append(value)
+
+        if operator in flattened:
+            for key, sublist in _lookup_operator.iteritems():
+                for val in sublist:
+                    if val==operator:
+                        properties['operator'] = key
+            return (shape, properties, fid)
+
+    return (shape, properties, fid)
```
Comment (okavvada) [inline vectordatasource/transform.py]: @rmarianski, is this what you were going for?

Comment (rmarianski) [inline vectordatasource/transform.py]: I was thinking that the lookup table would be created first, and then the implementation would just try to lookup the value. Something like the following (untested):

```python
_lookup_operator_rules = {
    'United States National Park Service': (
        'National Park Service',
        'US National Park Service',
        'U.S. National Park Service',
        'US National Park service'
    ),
    'United States Forest Service': (
        'US Forest Service',
        'U.S. Forest Service',
        'USDA Forest Service',
        'United States Department of Agriculture',
        'US National Forest Service',
        'United State Forest Service',
        'U.S. National Forest Service'
    ),
    'National Parks & Wildife Service NSW': (
        'Department of National Parks NSW',
        'Dept of NSW National Parks',
        'Dept of National Parks NSW',
        'Department of National Parks NSW',
        'NSW National Parks',
        'NSW National Parks & Wildlife Service',
        'NSW National Parks and Wildlife Service',
        'NSW Parks and Wildlife Service',
        'NSW Parks and Wildlife Service (NPWS)',
        'National Parks & Wildife Service NSW',
        'National Parks and Wildlife NSW',
        'National Parks and Wildlife Service NSW'
    ),
}
normalized_operator_lookup = {}
for normalized_operator, variants in _lookup_operator_rules.items():
    for variant in variants:
        normalized_operator_lookup[variant] = normalized_operator


def normalize_operator_values(shape, properties, fid, zoom):
    """
    There are many operator-related tags, including 'National Park Service',
    'U.S. National Park Service', 'US National Park Service' etc that refer
    to the same operator tag. This function promotes a normalized value
    for all alternatives in specific operator values.

    See https://github.com/tilezen/vector-datasource/issues/927.
    """

    operator = properties.get('operator', None)
    if operator is not None:
        normalized_operator = normalized_operator_lookup.get(operator, None)
        if normalized_operator:
            properties['operator'] = normalized_operator

    return shape, properties, fid
```

**Comment by okavvada** [comment]
review from @nvkelso in person.

**Pull request merged**